packages/wonder-blocks-forms/components/dropdown.js

```diff
@@ -0,0 +1,121 @@
+// @flow
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+
+import {View} from "wonder-blocks-core";
+
+type Props = {
+    onClose: () => void,
+    onOpen: () => void,
+    onChange: (e: SyntheticEvent<>, value: string) => void,
+    show: boolean,
+    selection: ?any,
+    controlled: boolean,
+    items?: Array<{
+        label: string,
+        value: any,
+    }>,
+};
+
+type State = {
+    show: boolean,
+    selection: ?any,
```

Comment:
It would be interesting to reuse this inside of props, e.g.
```
type State = {|
   show: boolean,
   selection: ?any,
|};

type Props = {|
   ...State,
   onClose: () => void,
   onOpen: () => void,
   ...
|};
```

Reply:
I think whenever the prop can be used to set an initial value, it should have a different name that suggests that (like `initialShow` instead of `show`).

My rationale for this is that we should make it clear that the prop only controls the initial value; changing the prop later on won't update the component's state.

But yeah that would foil the neat spreading state in props thing :/

Reply:
I think being explicit is worth the price.